src/main/java/echoServer/http/RequestParser.java

```diff
@@ -0,0 +1,14 @@
+package echoServer.HTTP;
+
+public class RequestParser {
+
+    public static HTTPRequest parser(String rawRequest) {
+
+        String[] rows = rawRequest.split("\r\n");
+        String[] firstRequestRow = rows[0].split(" ");
```

Comment:
Great single responsibility here! A few nits:  the method says `parser`, but the method is an action, so I reckon `parse` would be a better name.  And on lines 10 to 11, you can condense into just `return new HTTPRequest(...etc...)`